docs/source/guides/mod-config.md

```diff
@@ -1,166 +1,328 @@
+	config = myConfig,
+	defaultConfig = myDefaultConfig,
+	showDefaultSetting = true,
+})
+```
 
-If your mod adds any hotkey function, a way to rebind the hotkey in the mod config menu is a must.
+This will result in the default values of settings automatically being displayed underneath their descriptions, provided those settings are in a `SideBarPage`.
 
-!!! example "**MyMod\\main.lua**"
-	```lua
-	settings:createKeyBinder({
-		label = "Assign Keybind",
-		description = "Assign a new keybind to perform awesome tasks.",
-		variable = mwse.mcm.createTableVariable{ id = "keybind", table = config },
-		allowCombinations = true,
-	})
-	```
+### Separate config.lua
+
+As your default config gets long, you might want to create a separate Lua file for it.
+
+```lua linenums="1" title="My Awesome Mod/defaultConfig.lua"
+-- Define the defaultConfig in a separate file so we can access it later.
+local defaultConfig = {
+	enabled = true,
+	keybind = {
+		keyCode = tes3.scanCode.k,
+		isShiftDown = false,
+		isAltDown = false,
+		isControlDown = true,
+	}, -- Ctrl + K
+	nonEdible = {
+		["ingred_diamond_01"] = true,
+		["ingred_scrap_metal_01"] = true,
+	}
+}
+return defaultConfig
+```
+
+```lua linenums="1" title="My Awesome Mod/config.lua"
+local defaultConfig = require("My Awesome Mod.defaultConfig")
+local config = mwse.loadConfig("My Awesome Mod", defaultConfig)
+return config
+```
+
+We can then use the `require` function to import the `config` and `defaultConfig` in `main.lua`.
+
+```lua linenums="1" title="My Awesome Mod/main.lua"
+local config = require("My Awesome Mod.config")
+local defaultConfig = require("My Awesome Mod.defaultConfig")
+```
+
+!!! note
+	Having a dedicated config file is important when your mod has multiple files.
+	When using a dedicated config file, you can import it from every file of your mod using `require`, and all of those files
+	will use a sychronized copy of your config.
 
 ### Exclusions Page
 
-Let's say your My Awesome Mod makes certain ingredients non-edible. Using an Exclusions Page, you can allow users to manually add what ingredients should be non-edible in game.
-
-!!! example "**MyMod\\main.lua**"
-	```lua
-	template:createExclusionsPage({
-		label = "Non-Edible Ingredients",
-		description = "Ingredients that can't be directly consumed via equipping.",
-		leftListLabel = "Non-Edible Ingredients",
-		rightListLabel = "Ingredients",
-		variable = mwse.mcm.createTableVariable{
-			id = "nonEdible",
-			table = config,
+Let's say your Awesome Mod makes certain ingredients non-edible. Using an Exclusions Page, you can allow users to manually add what ingredients should be non-edible in game.
+
+```lua linenums="1" title="My Awesome Mod/main.lua"
+template:createExclusionsPage({
+	label = "Non-Edible Ingredients",
+	description = "Ingredients that can't be directly consumed via equipping.",
+	leftListLabel = "Non-Edible Ingredients",
+	rightListLabel = "Ingredients",
+	variable = mwse.mcm.createTableVariable{
```

Comment:
I would also like to use `configKey` here, but the `ExclusionsPage` does not yet support that syntax so I thought it best to keep the `createTableVariable` for now. 

But it would be nice to add support for the `configKey` to the `ExclusionsPage` soon.

Reply:
It's fine for now. We can update the guide after we add such support to the exclusions page.